lawsearch.html

```diff
@@ -164,52 +164,104 @@ <h1 class="pagetitle">Law&nbsp;Search</h1>
   "regexes": [/^Cal\.? (?<code>[a-z]+) §? ?(?<section>[\w\.]+)/i],
   "URLParts": {"baseURL": "https://leginfo.legislature.ca.gov/faces/codes_displaySection.xhtml?lawCode=[code]&sectionNum=[section]"},
   "forceUpperCase": ["code"]
-}]
+}];
 
 /*
-Checks the given query against the schemas, and uses the matched
-schema to generate a URL, then redirect to that URL. Returns true
-if successful, false otherwise. Outputs error messages to the
-DOM element with id "explainer".
+  Parses URL search query, and passes it to handleQuery().
+  Runs upon page load, while the LawSearch page is hidden.
+  Returns true if the page should be unhidden, i.e. upon a *failed*
+  search. Returns false otherwise.
+  Also puts the query into the search bar.
 */
-function lookup(query) {
+document.addEventListener("DOMContentLoaded", () => {
+  if (!location.search) {
+    return document.body.removeAttribute('hidden');
+  }
+
+  let query = decodeURIComponent(location.search).trim().replace(/^\?(?:q=)?|\.$|,$|;$/g, '');
+  document.getElementById("q").value = query.replace(/\+/g, ' ');
+
+  handleQuery(query);
+});
+
+function handleSearch(event) {
+  event.preventDefault()
+  const query = document.getElementById("q").value
+  handleQuery(query);
+}
+
+function handleQuery(query) {
```

Comment:
Dom interactions were pulled into the top level functions handleSearch, handleQuery, urlQuery to isolate the url building code from the dom interactions. This separates our concerns.